pkg/operator/ceph/object/spec.go

```diff
@@ -678,3 +764,34 @@ func (c *clusterConfig) rgwTLSSecretType(secretName string) (v1.SecretType, erro
 func getDaemonName(rgwConfig *rgwConfig) string {
 	return fmt.Sprintf("ceph-%s", generateCephXUser(rgwConfig.ResourceName))
 }
+
+func (c *clusterConfig) generateVaultAgentConfigMap() error {
+	k := k8sutil.NewConfigMapKVStore(c.store.Namespace, c.context.Clientset, c.ownerInfo)
+	if _, err := k.GetValue(c.clusterInfo.Context, vaultAgentCM, vaultAgentConfigFile); err == nil || !kerrors.IsNotFound(err) {
+		logger.Infof("vault agent config map %q for object store %q already exists, not overwriting", vaultAgentCM, c.store.Name)
```

Comment:
If the connection details are updated in the object store CR, shouldn't this configmap be updated?

Reply:
Most likely the RGW pod needs to restart if we update connection details. Otherwise, the sidecar pod does not have the latest details. Apart from config map a lot of environment variables are set for the side car pod based on the connection details

Reply:
Thanks now I got it, during I need to update the configmap as well, currently only creating the cm. Will update in the next version